test/vcr_clj/test/core.clj

```diff
@@ -155,3 +155,32 @@
           (.join))
         (is (= 2 (count (calls increment))))
         (is (= 100 @p))))))
+
+
+(deftest middleware-test
+  (defn args-test [args]
+    (is (= '(2 3) args)))
+
+  (defn res-test [res]
+    (is (= 5 res)))
+
+  (defn sample-middleware [vcr-clj-fn]
+    (fn [& args]
+      (args-test args)
+      (let [res (apply vcr-clj-fn args)]
+        (res-test res)
+        res)))
```

Comment:
could you either move these `defn`s to the top level or transform them into a `let` or `letfn`? I don't think nested `def`s are idiomatic in clojure.